tests/python/testing.py

```diff
@@ -206,16 +207,23 @@ def get_sparse():
 
 
 @strategies.composite
-def _dataset_and_weight(draw):
+def _dataset_weight_margin(draw):
     data = draw(_unweighted_datasets_strategy)
     if draw(strategies.booleans()):
         data.w = draw(arrays(np.float64, (len(data.y)), elements=strategies.floats(0.1, 2.0)))
+    if draw(strategies.booleans()):
```

Comment:
Why duplicate an `if` condition?

Reply:
The samples from a random variable, so we can get four possible combinations of DMatrix with weights and base_margin on/off.